fenix/app/src/test/java/org/mozilla/fenix/debugsettings/DefaultDebugSettingsCacheTest.kt

```diff
@@ -0,0 +1,54 @@
+package org.mozilla.fenix.debugsettings
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import kotlinx.coroutines.flow.drop
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.test.runTest
+import mozilla.components.support.test.robolectric.testContext
+import org.junit.After
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.debugsettings.data.DefaultDebugSettingsCache
+
+@RunWith(AndroidJUnit4::class)
+class DefaultDebugSettingsCacheTest {
+
+    @After
+    fun cleanUp() = runTest {
+        val defaultDebugSettingsCache = DefaultDebugSettingsCache(
+            context = testContext,
+            writeScope = this,
+        )
+        defaultDebugSettingsCache.clearPreferences()
+    }
+
+    @Test
+    fun `GIVEN the debug drawer is disabled WHEN the flag is made true THEN the store should have the correct value`() = runTest {
+        val flowEmissionsToIgnore = 2
+        val defaultDebugSettingsCache = DefaultDebugSettingsCache(
+            context = testContext,
+            writeScope = this,
+        )
+        defaultDebugSettingsCache.setDebugDrawerEnabled(false)
+
+        defaultDebugSettingsCache.setDebugDrawerEnabled(true)
+
+        assertTrue(defaultDebugSettingsCache.debugDrawerEnabledFlow.drop(flowEmissionsToIgnore).first())
+    }
+
+    @Test
+    fun `GIVEN the debug drawer is enabled WHEN the flag is made false THEN the store should have the correct value`() = runTest {
+        val flowEmissionsToIgnore = 2
```

Comment:
nit: this could be moved closer to the call site - same applies to the other `flowEmissionsToIgnore`

Reply:
I think this makes sense! I was also struggling to find a way to cover this, since `flow.count()` was hanging the test, so I'll give this a try!